py/metis/site_activation_vis.py

```diff
@@ -33,12 +33,16 @@
 in the input files).
 """
 
-def site_activation(original_ville):
+def site_activation(original_ville, initial_proposed_activation=None):
     """Make the site activation display.
 
     Args:
         original_ville: An xr.Dataset containing the ville as it is now.
             This is what all changes will be compared against.
+        initial_proposed_activation: Optional. An xr.DataArray with dimensions
```

Comment:
I worry that this change adds complexity to this method without really relieving the caller of any responsibility. Why use this instead of just setting original_ville['site_activation'] before calling this method? Put differently, why allow overwriting site_activation here, but not allow overwriting of other things (site_capacity, participant_fraction, trial_size_cap, etc.)?

Reply:
This vis only allows the user to toggle site_activations by clicking the buttons. So by putting in an option to specify the proposed activation, it's equivalent to clicking a button for each site. This option saves the trial planner a lot of time, as they don't have to wait for the vis to reload after each click.

Reply:
I agree that we want to give the option to batch update, but it doesn't make sense to me for it to be an argument to this function. It's the same amount of work for the user to just set the data_var in the ville before calling the old version of the function. That approach is more flexible (because other trial plan parameters can be updated that way) and allows this function to be simpler. Is something unsatisfying about that workflow?

Reply:
If they set the data_var in the ville, then that becomes the 'original' plan. Currently the only way to make the original and proposed villes different is through the button interface. This PR changes that.

Reply:
Ah, I see. The version before the modification is recorded by set_up_dataset.